src/test/java/com/wolox/training/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,175 @@
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    void setUp() {
+        user = new User("jhon_doe", "JhonDoe", LocalDate.parse("1990-01-01"));
+    }
+
+    @Test
+    public void givenUsers_whenGetUsers_thenReturnsJsonArray() throws Exception {
+
+        List<User> allUsers = Collections.singletonList(user);
+
+        given(userRepository.findAll()).willReturn(allUsers);
+
+        mockMvc.perform(get(URI.create("/api/users"))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$", hasSize(1)))
+            .andExpect(jsonPath("$[0].name", is(user.getName())));
+    }
+
+    @Test
+    public void givenUsers_whenGetUser_thenReturnsJson() throws Exception {
+
+        given(userRepository.findById(user.getId())).willReturn(ofNullable(user));
+
+        mockMvc.perform(get(URI.create(String.format("/api/users/%d", user.getId())))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.name", is(user.getName())));
+    }
+
+    @Test
+    public void givenValidUser_whenPostUser_thenReturnsJson() throws Exception {
+
+        given(userRepository.save(user)).willReturn(user);
+
+        mockMvc.perform(post(URI.create("/api/users"))
+            .content(objectMapper.writeValueAsString(user))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isCreated());
+    }
+
+    @Test
+    public void givenInvalidUser_whenPostUser_thenReturnsBadRequest() throws Exception {
+        user = new User();
+        user.setName("Jane Doe");
+        user.setUsername("jane_doe");
+        given(userRepository.save(user)).willReturn(user);
+
+        mockMvc.perform(post(URI.create("/api/users"))
+            .content(objectMapper.writeValueAsString(user))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void givenValidUser_whenDeletesUser_thenReturnsNoContent() throws Exception {
+
+        given(userRepository.findById(user.getId())).willReturn(ofNullable(user));
+
+        mockMvc.perform(delete(URI.create(String.format("/api/users/%d", user.getId())))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isNoContent());
+    }
+
+    @Test
+    public void givenValidUser_whenUpdatesUser_thenReturnsAccepted() throws Exception {
+        given(userRepository.findById(user.getId())).willReturn(ofNullable(user));
+
+        String new_name = "Jane Doe";
+        user.setName(new_name);
+
+        mockMvc.perform(put(URI.create(String.format("/api/users/%d", user.getId())))
+            .content(objectMapper.writeValueAsString(user))
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isAccepted());
+    }
+
+    @Test
+    public void givenAuthor_whenFindTopByName_thenReturnsJson() throws Exception {
```

Comment:
What happens if no user is found with the passed name? 🤔(I'm really asking, but having a test for that would have answered it 😋). Is the current behaviour what you expected to be?